token_reject_transaction_e2e_test.go

```diff
@@ -0,0 +1,960 @@
+	frozenTx, err := NewTransferTransaction().
+		AddApprovedTokenTransfer(tokenID, receiver, -5, true).
+		AddTokenTransfer(tokenID, spender, 5).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+	transfer, err := frozenTx.SignWith(spenderKey.PublicKey(), spenderKey.Sign).Execute(env.Client)
+	require.NoError(t, err)
+	_, err = transfer.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	env.Client.SetOperator(env.OperatorID, env.OperatorKey)
+
+	// reject the token
+	frozenTxn, err := NewTokenRejectTransaction().
+		SetOwnerID(receiver).
+		AddTokenID(tokenID).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+	resp, err := frozenTxn.Sign(key).Execute(env.Client)
+	require.NoError(t, err)
+	_, err = resp.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// verify the allowance - should be 0 , because the receiver is no longer the owner
+	env.Client.SetOperator(spender, spenderKey)
+	frozenTx, err = NewTransferTransaction().
+		AddApprovedTokenTransfer(tokenID, receiver, -5, true).
+		AddTokenTransfer(tokenID, spender, 5).FreezeWith(env.Client)
+	tx, err = frozenTx.Sign(spenderKey).Execute(env.Client)
+	require.NoError(t, err)
+	_, err = tx.SetValidateStatus(true).GetReceipt(env.Client)
+	require.ErrorContains(t, err, "SPENDER_DOES_NOT_HAVE_ALLOWANCE")
+	env.Client.SetOperator(env.OperatorID, env.OperatorKey)
+
+	// same test for nft
+
+	// create nft with treasury
+	nftID := createNftHelper(t, &env)
+
+	// mint
+	mint, err := NewTokenMintTransaction().SetTokenID(nftID).SetMetadatas(initialMetadataList).Execute(env.Client)
+	receipt, err := mint.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	serials := receipt.SerialNumbers
+
+	// transfer nfts to the receiver
+	tx, err = NewTransferTransaction().
+		AddNftTransfer(nftID.Nft(serials[0]), env.Client.GetOperatorAccountID(), receiver).
+		AddNftTransfer(nftID.Nft(serials[1]), env.Client.GetOperatorAccountID(), receiver).
+		AddNftTransfer(nftID.Nft(serials[2]), env.Client.GetOperatorAccountID(), receiver).
+		AddNftTransfer(nftID.Nft(serials[3]), env.Client.GetOperatorAccountID(), receiver).
+		Execute(env.Client)
+	require.NoError(t, err)
+	_, err = tx.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// approve allowance to the spender
+	frozenApprove, err = NewAccountAllowanceApproveTransaction().
+		ApproveTokenNftAllowance(nftID.Nft(serials[0]), receiver, spender).
+		ApproveTokenNftAllowance(nftID.Nft(serials[1]), receiver, spender).
+		FreezeWith(env.Client)
+	require.NoError(t, err)
+	approve, err = frozenApprove.Sign(key).Execute(env.Client)
+	require.NoError(t, err)
+	_, err = approve.SetValidateStatus(true).GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	// verify the spender has allowance
+	env.Client.SetOperator(spender, spenderKey)
+	frozenTx, err = NewTransferTransaction().
+		AddApprovedNftTransfer(nftID.Nft(serials[0]), receiver, spender, true).
```

Comment:
I think here should be `AddApprovedNftTransfer(nftID.Nft(serials[0]), spender, receiver, true)`, as `spender` should send nft.